From linux-usb-devel-bounces@lists.sourceforge.net Wed Nov  8 06:35:24 2006
From: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
To: Andrew Morton <akpm@osdl.org>
Date: Wed, 8 Nov 2006 15:33:38 +0100
MIME-Version: 1.0
Content-Disposition: inline
Message-Id: <200611081533.41048.m.kozlowski@tuxland.pl>
Cc: Greg KH <greg@kroah.com>, linux-kernel@vger.kernel.org, linux-usb-devel@lists.sourceforge.net
Subject: usb: writing_usb_driver free urb cleanup


	Allright. As Greg KH suggested I split this big patch into smaller ones to
make the changes easier to review. Having no better idea how to split that I 
split it on a 'patch per file' basis. All those patches clean redundant 'if' before 
usb_unlink/free/kill_urb():

if (urb)
	usb_free_urb(urb); /* unlink / free / kill */

I decided not to touch bigger 'if's like 

if (urb) {
	usb_kill_urb(urb);
	usb_free_urb(urb);
	urb = NULL;
}

as that would be probably too intrusive. One of patches also fixes 
drivers/usb/misc/auerswald.c memleak I found when digging the code. All those
patches are against 2.6.19-rc4.

Signed-off-by: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 Documentation/DocBook/writing_usb_driver.tmpl |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- gregkh-2.6.orig/Documentation/DocBook/writing_usb_driver.tmpl
+++ gregkh-2.6/Documentation/DocBook/writing_usb_driver.tmpl
@@ -345,8 +345,7 @@ static inline void skel_delete (struct u
         usb_buffer_free (dev->udev, dev->bulk_out_size,
             dev->bulk_out_buffer,
             dev->write_urb->transfer_dma);
-    if (dev->write_urb != NULL)
-        usb_free_urb (dev->write_urb);
+    usb_free_urb (dev->write_urb);
     kfree (dev);
 }
   </programlisting>
